nucypher/blockchain/eth/signers.py

```diff
@@ -15,17 +19,38 @@ class Signer(ABC):
 
     class InvalidSignerURI(ValueError):
         """Raised when an invalid signer URI is detected"""
+        def __init__(self, uri=None, text=None):
```

Comment:
It feels like we should just have the creator of the `InvalidSignerURI` provide the full text for the error, instead of having logic to incorporate the `uri` i.e. I don't think we should have the `uri` parameter for the function.

Reply:
As one would not raise an InvalidSignerURI unless there were an invalid URI to complain about, having the ctor take the uri makes sense.

Text formatting is not necessary and in fact the base exception class stuff all of the arguments into self.args anyways which appear when the exception is printed (i.e. __str__() returns them), so have added *args / **kwargs to __init__() and left it that.

Reply:
My hesitation was that InvalidSignerURI constructor allowed `uri` to default `None` which then leads to extra logic within `__init__` for checking whether the `uri` is set or not. If that is your expectation then `uri` should not default to `None` but rather we should force a value to be provided.

If `uri` is indeed forced to be passed, then the logic in `__init__` can be boiled down to:
```python
    text = f"'{uri}': {text or 'is not a valid signer URI'}"
    args.append(text)
    super().__init__(*args)
```

Reply:
Latest commits takes a minimilist approach - any exceptions raised in Signer.from_signer_uri() are wrapped in an InvalidSignerURI (unless it's an InvalidSignerURI already).

As far as Ursula goes, all CryptoError and ValueError exceptions are caught in actions.make_cli_character() which just displays a red line with no information from the exception which is perhaps a bit too brief - however that's out of scope of this PR.

As for InvalidSignerURI, it possible to set attributes such as `uri` and `inner_exc` on the exception object so upstream handlers can inspect them but as none currently do, I've gone with minimal.